src/consensus/aft/raft.h

```diff
@@ -1103,26 +1106,49 @@ namespace aft
           return;
         }
 
-        state->last_idx = i;
-
-        Term sig_term = 0;
-        Index sig_index = 0;
-        auto tx = store->create_tx();
-        kv::DeserialiseSuccess deserialise_success;
-        ccf::PrimarySignature sig;
-        kv::ConsensusHookPtrs hooks;
-        if (consensus_type == ConsensusType::BFT)
+        auto ds = store->deserialise(entry, consensus_type, public_only);
+        if (ds == nullptr)
         {
-          deserialise_success = store->deserialise_views(
-            entry, hooks, public_only, &sig_term, &sig_index, &tx, &sig);
+          LOG_DEBUG_FMT("failed to deserialize we failed to apply");
+          send_append_entries_response(
+            r.from_node, AppendEntriesResponseType::FAIL);
+          return;
         }
-        else
+        append_entries.push_back(std::make_tuple(std::move(ds), i));
+      }
+
+      for (auto& ae : append_entries)
+      {
+        auto& [ds, i] = ae;
+        state->last_idx = i;
+
+        kv::DeserialiseSuccess deserialise_success =
+          kv::DeserialiseSuccess::FAILED;
+        if (ds != nullptr)
         {
-          deserialise_success =
-            store->deserialise(entry, hooks, public_only, &sig_term);
+          deserialise_success = ds->Execute();
```

Comment:
It's probably time to rename deserialise() to apply(), because we're clearly doing much more than deserialisation when the replicas re-execute.